src/colmap/geometry/rigid3.h

```diff
@@ -112,6 +112,17 @@ inline Eigen::Matrix6d GetCovarianceForRigid3dInverse(
   return adjoint_inv * covar * adjoint_inv.transpose();
 }
 
+// Calculate covariance (6 x 6) for relative transformation T_1^{-1}T_2.
+inline Eigen::Matrix6d GetCovarianceForRelativeRigid3d(
+    const Rigid3d& pose1,
+    const Rigid3d& pose2,
+    const Eigen::Matrix<double, 12, 12>& covar) {
```

Comment:
It will be good to provide some more documentation for this and specify what the covariance matrix contains. I also suggest using the a_from_b notation. For example:
```suggestion
// Returns covariance for relative pose a_from_c. Expects covariance as ...
inline Eigen::Matrix6d GetCovarianceForRelativeRigid3d(
    const Rigid3d& b_from_a,
    const Rigid3d& b_from_c,
    const Eigen::Matrix<double, 12, 12>& cov) {
```

Reply:
This should not matter. I believe the math works for both cam_from_world and world_from_cam. So its really the composition of rigid3d as long as the covariance is in the same convention of how the rigid3d is defined.

Reply:
Also I just realized that the second pose does not contribute to the covariance propagation and is thus not required. I have made a simplification and made some improvements to the comments: https://github.com/colmap/colmap/pull/3168/commits/1f8ed3c17ade78e3a7fe5cbbe9595133c0c30fc2

Reply:
What I wanted to imply is that if the left hand side of the two transformations is not the same, then the resulting covariance is not meaningful. It then also matters how the covariance is specified. In the case where the input poses are b_from_an and b_from_c, it's the covariance blocks b_from_a_cov and b_from_c_cov.

For our use case, I am also thinking that it may be more useful to design the interface in the other direction where the relative pose in your notation is composed as T_1 * T_2^{-1}. I assume we'll use this for computing relative pose covariances and we will always have cam_from_world poses as input?

Reply:
You are absolutely right. I updated the notation to T_2 * T_1^{-1} and did the derivation again. The tests are also updated. I also changed the convention to a_from_c and b_from_c and the relative pose will be b_from_a.

Reply:
PTAL

Reply:
One thing to mention: with the updated convention the second pose does contribute to the covariance propagation, so I added it back to the interface.